salt/client/ssh/ssh_py_shim.py

```diff
@@ -273,6 +274,11 @@ def main(argv):  # pylint: disable=W0613
     """
     Main program body
     """
+    try:
+        lock_fh = open("{0}.lock".format(OPTIONS.saltdir), "w")
+        fcntl.flock(lock_fh.fileno(), fcntl.LOCK_EX)
```

Comment:
Can you please leave comments in the code that answer the two questions below?

1) what is the problem that is fixed by adding this lock
2) how is adding the lock fixing the problem


By the way, you don't need to call `.fileno()` if you don't want to :)

Reply:
`fileno()` call removed and comments appended with https://github.com/openSUSE/salt/pull/414/commits/4e54a909c856e0852cb8dbc08f6fcc8e9a3c66ed

Reply:
Thanks for the comments. It's definitely better now